net/strongswan/files/swanctl.init

```diff
@@ -0,0 +1,604 @@
+	ike)
+		;;
+	ikev1)
+		swanctl_xappend2 "version = 1" ;;
+	ikev2)
+		swanctl_xappend2 "version = 2" ;;
+	*)
+		warning "Keyexchange $keyexchange not supported"
+		keyexchange=
+		;;
+	esac
+
+	[ $mobike -eq 1 ] && swanctl_xappend2 "mobike = yes" || swanctl_xappend2 "mobike = no"
+
+	if [ -n "$rekeytime" ]; then
+		swanctl_xappend2 "rekey_time = $rekeytime"
+
+		if [ -z "$overtime" ]; then
+			overtime=$(seconds2time $(($(time2seconds $rekeytime) / 10)))
+		fi
+	fi
+	[ -n "$overtime" ] && swanctl_xappend2 "over_time = $overtime"
+
+	swanctl_xappend2 "proposals = $ike_proposal"
+	[ -n "$dpddelay" ] && swanctl_xappend2 "dpd_delay = $dpddelay"
+	[ "$keyingtries" = "%forever" ] && swanctl_xappend2 "keyingtries = 0" || swanctl_xappend2 "keyingtries = $keyingtries"
+
+	swanctl_xappend1 "}"
+	swanctl_xappend0 "}"
+
+	if [ "$auth_method" = pubkey ]; then
+		swanctl_xappend0 ""
+
+		swanctl_xappend0 "secrets {"
+		swanctl_xappend1 "rsa {"
+		swanctl_xappend2 "filename = $local_key"
+		swanctl_xappend1 "}"
+		swanctl_xappend0 "}"
+
+		swanctl_xappend0 ""
+
+		if [ -n "$ca_cert" ]; then
+			swanctl_xappend0 "authorities {"
+			swanctl_xappend1 "$config_name {"
+			swanctl_xappend2 "cacert = $ca_cert"
+			swanctl_xappend1 "}"
+			swanctl_xappend0 "}"
+		fi
+
+	elif [ "$auth_method" = psk ]; then
+		swanctl_xappend0 ""
+
+		swanctl_xappend0 "secrets {"
+		swanctl_xappend1 "ike {"
+		swanctl_xappend2 "secret = $pre_shared_key"
+		if [ -z "$local_id" ]; then
+			swanctl_xappend2 "id1 = $local_id"
+			if [ -z "$remote_id" ]; then
+				swanctl_xappend2 "id2 = $remote_id"
+			fi
+		fi
+	else
+		warning "AuthenticationMode $auth_mode not supported"
+	fi
+
+	swanctl_xappend0 ""
+}
+
+do_preamble() {
+	swanctl_xappend0 "# generated by /etc/init.d/swanctl"
+}
+
+config_ipsec() {
+	local debug
+	local rtinstall_enabled
+	local routing_tables_ignored
+	local routing_table
+	local routing_table_id
+	local interface
+	local device_list
+
+	swan_reset
+	swanctl_reset
+	do_preamble
+
+	config_get debug "$1" debug 0
+	config_get_bool rtinstall_enabled "$1" rtinstall_enabled 1
+	[ $rtinstall_enabled -eq 1 ] && install_routes=yes || install_routes=no
+
+	# prepare extra charon config option ignore_routing_tables
+	for routing_table in $(config_get "$1" "ignore_routing_tables"); do
+		if [ "$routing_table" -ge 0 ] 2>/dev/null; then
+			routing_table_id=$routing_table
+		else
+			routing_table_id=$(sed -n '/[ \t]*[0-9]\+[ \t]\+'$routing_table'[ \t]*$/s/[ \t]*\([0-9]\+\).*/\1/p' /etc/iproute2/rt_tables)
+		fi
+
+		[ -n "$routing_table_id" ] && append routing_tables_ignored "$routing_table_id"
+	done
+
+	local interface_list=$(config_get "$1" "interface")
+	if [ -z "$interface_list" ]; then
+		WAIT_FOR_INTF=0
+	else
+		for interface in $interface_list; do
+			network_get_device device $interface
+			[ -n "$device" ] && append device_list "$device" ","
+		done
+		[ -n "$device_list" ] && WAIT_FOR_INTF=0 || WAIT_FOR_INTF=1
+	fi
+
+	swan_xappend0 "# generated by /etc/init.d/swanctl"
```

Comment:
We shouldn't generate the default file contents again, just the ones that are not available to the user to edit at any time (so effectively just the settings we want).
That way the user can decide if he/she wants to include the default config files, or not.
Effectively that means removing load_modular and the the plugins include line from here.

Reply:
We're generating the entire profile via UCI, so the user doesn't have to hand-edit anything.

Reply:
I am talking about that the file we're creating is `/var/ipsec/strongswan.conf` and we're adding `"\ninclude /var/ipsec/strongswan.conf"` to `/etc/strongswan.conf`.
`/etc/strongswan.conf` default content already contains `charon.load_modular=yes` and the include line for `/etc/strongswan.d/*.conf`.
`/etc/strongswan.d/charon.conf` effectively contains the following line, too, already:
```
charon {
plugins {
include /etc/strongswan.d/charon/*.conf
}
```

If the user doesn't want that, he/she removed that. And then we add that back in via `/var/ipsec/strongswan.conf`. I think the user knows better in this regard.
We should just remove these includes from our generated file, especially given that we don't use any of the files that are included by the include directives in the generated file.

Reply:
Okay, removed:

```
  load_modular = yes
  ...
  plugins {
    include /etc/strongswan.d/charon/*.conf
  }
```